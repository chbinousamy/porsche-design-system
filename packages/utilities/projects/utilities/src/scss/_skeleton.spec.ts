import * as path from 'path';
import * as fs from 'fs';
import * as sass from 'sass';

const borderVariables = fs.readFileSync(path.resolve('./src/scss/lib/_border.scss'), 'utf8');
const themeVariables = fs.readFileSync(path.resolve('./src/scss/lib/_theme.scss'), 'utf8');
const skeletonMixin = fs.readFileSync(path.resolve('./src/scss/_skeleton.scss'), 'utf8').replace(/@import.*;/g, '');

describe('pds-skeleton()', () => {
  it.each([
    {},
    { theme: 'light', borderRadius: 'small' },
    { theme: 'light', borderRadius: 'medium' },
    { theme: 'light', borderRadius: '6px' },
    { theme: 'dark', borderRadius: 'small' },
    { theme: 'dark', borderRadius: 'medium' },
  ])('should return correct css for opts: %s', (opts) => {
    const result = sass.compileString(`
${borderVariables} ${themeVariables} ${skeletonMixin} div {
  @include pds-skeleton(${Object.values(opts).join(', ')});
}`);
    expect(result.css).toMatchSnapshot();
  });
});
