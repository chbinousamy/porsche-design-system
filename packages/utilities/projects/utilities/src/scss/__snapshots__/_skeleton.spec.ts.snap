// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pds-skeleton() should return correct css for opts: { borderRadius: '6px' } 1`] = `
"div {
  border-radius: 6px;
  display: block;
  background: #eeeff2 linear-gradient(to right, transparent 0%, #f7f7f7 20%, transparent 50%) 0 0/200% 100%;
  animation: skeletonAnimation var(--transition-duration-long, 0.6s) var(--easing-base, cubic-bezier(0.25, 0.1, 0.25, 1)) infinite;
}
@keyframes skeletonAnimation {
  from {
    background-position: 100% 0;
  }
  to {
    background-position: -100% 0;
  }
}"
`;

exports[`pds-skeleton() should return correct css for opts: { borderRadius: 'medium' } 1`] = `
"div {
  border-radius: 8px;
  display: block;
  background: #eeeff2 linear-gradient(to right, transparent 0%, #f7f7f7 20%, transparent 50%) 0 0/200% 100%;
  animation: skeletonAnimation var(--transition-duration-long, 0.6s) var(--easing-base, cubic-bezier(0.25, 0.1, 0.25, 1)) infinite;
}
@keyframes skeletonAnimation {
  from {
    background-position: 100% 0;
  }
  to {
    background-position: -100% 0;
  }
}"
`;

exports[`pds-skeleton() should return correct css for opts: { borderRadius: 'small' } 1`] = `
"div {
  border-radius: 4px;
  display: block;
  background: #eeeff2 linear-gradient(to right, transparent 0%, #f7f7f7 20%, transparent 50%) 0 0/200% 100%;
  animation: skeletonAnimation var(--transition-duration-long, 0.6s) var(--easing-base, cubic-bezier(0.25, 0.1, 0.25, 1)) infinite;
}
@keyframes skeletonAnimation {
  from {
    background-position: 100% 0;
  }
  to {
    background-position: -100% 0;
  }
}"
`;

exports[`pds-skeleton() should return correct css for opts: {} 1`] = `
"div {
  border-radius: 4px;
  display: block;
  background: #eeeff2 linear-gradient(to right, transparent 0%, #f7f7f7 20%, transparent 50%) 0 0/200% 100%;
  animation: skeletonAnimation var(--transition-duration-long, 0.6s) var(--easing-base, cubic-bezier(0.25, 0.1, 0.25, 1)) infinite;
}
@keyframes skeletonAnimation {
  from {
    background-position: 100% 0;
  }
  to {
    background-position: -100% 0;
  }
}"
`;
